arch/configure.defaults

```diff
@@ -1544,8 +1544,8 @@ BYTESWAPIO      =       -byteswapio
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO) $(OMP)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     -module $(WRF_SRC_ROOT_DIR)/main
-TRADFLAG        =      CONFIGURE_TRADFLAG
-CPP             =      /lib/cpp CONFIGURE_CPPFLAGS
+TRADFLAG        =      
+CPP             =      $(SCC) -E 
```

Comment:
I do not understand why only three locations have been targeted for modification. Should every PGI build use the new option that is suggested? For example, I see ten locations where there are PGI stanzas in the configure.defaults file.
```
>grep -n PGI configure.defaults | grep ARCH
133:#ARCH    Linux x86_64 ppc64le i486 i586 i686, PGI compiler with gcc # serial smpar dmpar dm+sm
176:#ARCH    Linux x86_64 ppc64le, PGI compiler with pgcc, SGI MPT # serial smpar dmpar dm+sm
219:#ARCH    Linux x86_64 ppc64le, PGI accelerator compiler with gcc # serial smpar dmpar dm+sm
802:#ARCH    Darwin (MACOS) PGI compiler with pgcc #serial smpar dmpar dm+sm
1210:#ARCH    Cray XC CLE/Linux x86_64, PGI compiler with gcc # serial dmpar smpar dm+sm
1514:#ARCH    Linux x86_64 ppc64le i486 i586 i686, PGI compiler with pgcc # serial smpar dmpar dm+sm
1557:#ARCH    CYGWIN_NT i686, PGI compiler on Windows # serial smpar dmpar dm+sm
1614:#ARCH    Linux x86_64 ppc64le i486 i586 i686, PGI compiler with gcc -f90= # serial smpar dmpar dm+sm
1657:#ARCH    Darwin (MACOS) PGI compiler with pgcc -f90= #serial smpar dmpar dm+sm
1789:#ARCH    Linux x86_64 ppc64le i486 i586 i686, PGI compiler with pgcc -f90= # serial smpar dmpar dm+sm
```

Reply:
I only changed the pure Linux x86_64 ones because I don't have access to those other platforms. I assume that similar changes should be made to those other cases, but cannot verify. Basically I'm just erring on the side of caution. 

Anything which is a hybrid PGI/GCC configuration I also left alone because it was outside the scope of what I was trying to get to work. For those hybrids, I don't see a similar option on the gcc compiler to stop at the preprocessing step, hence one needs to run the c preprocessor directly. It's probably worth adding another environment variable for the configure script to pay attention to, leaving the default (if it's not set) to be /lib/cpp.

I hadn't solved the gnu side of things, leaving that for some future pull request.

Reply:
> I don't see a similar option on the gcc compiler to stop at the preprocessing step

From https://gcc.gnu.org/onlinedocs/gcc-8.2.0/gcc/Overall-Options.html#Overall-Options:
> gcc -E
>    Stop after the preprocessing stage; do not run the compiler proper. 
>    The output is in the form of preprocessed source code, which is sent to the standard output.
>
>    Input files that don’t require preprocessing are ignored. 
so using the same definition of CPP would also work there.  This would also include the compiler-specific include directories, but I don't think those are relevant for WRF.

It looks like Intel uses the same flag with the same meaning, as well as a -EP flag mirroring gcc -E -P:
ifort: https://software.intel.com/en-us/fortran-compiler-developer-guide-and-reference-ep
icc: https://software.intel.com/en-us/node/523064

It looks like you plan to leave other compilers for a later pull request, so this will probably serve mostly as a reference for that future pull request.